### PR TITLE
[BUGFIX] Fix inconsistent returns

### PR DESCRIPTION
**Pull request by cedricziel**
Also simplifies cache retrieval with a class local cache instance.

### CODE REVIEW

File: Classes/Backend/DynamicFlexForm.php [hunk 1]
```diff
@@ -323,7 +320,10 @@ protected function recursivelyEvaluateClosures(array $dataStructureArray, array
      */
     protected function getCache()
     {
-        return GeneralUtility::makeInstance(CacheManager::class)->getCache('flux');
+        if ($this->cache !== null) {
```
Comment (NamelessCoder) [inline Classes/Backend/DynamicFlexForm.php]: This happens once per field and at most a couple of times; cachemanager already caches the cache so we don't really win anything by this.

File: Classes/Backend/DynamicFlexForm.php [hunk 2]
```diff
@@ -78,9 +78,6 @@ public function injectRecordService(WorkspacesAwareRecordService $recordService)
         $this->recordService = $recordService;
     }
 
-    /**
```
Comment (NamelessCoder) [inline Classes/Backend/DynamicFlexForm.php]: Leave this in please - methods without doc comments not allowed.

File: Classes/Backend/DynamicFlexForm.php (hunk 1 above)
Comment (cedricziel) [inline Classes/Backend/DynamicFlexForm.php]: It does, but we save a bunch of calls in GeneralUtility as well. Your choice - I'd be happy to remove it if you think it's better without a local cache.

Comment (NamelessCoder) [inline Classes/Backend/DynamicFlexForm.php]: convert it to a static memoizer in the function itself and we've got a deal ;)

Comment (cedricziel) [inline Classes/Backend/DynamicFlexForm.php]: I don't think I get what you want to memoize here. The values or a single cache instance on a static property?

Comment (cedricziel) [inline Classes/Backend/DynamicFlexForm.php]: First encounter with a scoped static field. Thx for fixing it :)

**Pull request merged**